earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -18,19 +18,139 @@
 #include <unistd.h>
 #include <sys/syscall.h>
 #include <time.h>
-
+#include <vector>
 #include "common/performancelogger.h"
 #include "common/notify.h"
 #include "common/timeutils.h"
 
 using namespace std;
 using namespace getime;
 
-// Initialize static members of Profiler class
-PerformanceLogger * const PerformanceLogger::_instance = new PerformanceLogger();
+std::ofstream ioPrefFile;
+std::string ioFileName;
+std::ofstream timePrefFile;
+std::ofstream threadPrefFile;
+
+void checkForLeadingZero(
+        std::stringstream& io,
+        std::stringstream& thread,
+        std::stringstream& time,
+        const int& val) {
+    if (val > 9) return;
+    std::string temp("0");
+    io << temp;
+    thread << temp;
+    time << temp;
+}
+
+void printToSStream(
+        std::stringstream& io,
+        std::stringstream& thread,
+        std::stringstream& time,
+        const char& delimiter,
+        const int& val) {
+    io << val << delimiter;
+    thread << val << delimiter;
+    time << val << delimiter;
+}
+
+void openFiles() {
+    time_t t = time(0);
+    tm date = *localtime(&t);
+    std::stringstream io,time,thread;
+    io << "io.";
+    time << "time.";
+    thread << "thread.";
+    checkForLeadingZero(io,thread,time,date.tm_mon);
+    printToSStream(io,thread,time,'-',date.tm_mon);
+    checkForLeadingZero(io,thread,time,date.tm_mday);
+    printToSStream(io,thread,time,'-',date.tm_mday);
+    printToSStream(io,thread,time,'-',date.tm_year+1900);
+    checkForLeadingZero(io,thread,time,date.tm_hour);
+    printToSStream(io,thread,time,':',date.tm_hour);
+    checkForLeadingZero(io,thread,time,date.tm_min);
+    printToSStream(io,thread,time,':',date.tm_min);
+    checkForLeadingZero(io,thread,time,date.tm_sec);
+    printToSStream(io,thread,time,'.',date.tm_sec);
+    io << "csv";
+    thread << "csv";
+    time << "csv";
+    // if not open, open and add header
+    if (!ioPrefFile.is_open()) { 
+      ioPrefFile.open(io.str().c_str());
+      ioFileName = io.str();
+      ioPrefFile << "operation,object,startTime,endTime,bufferSize" << endl;
+    }
+    if (!timePrefFile.is_open()) {
+      timePrefFile.open(time.str().c_str());
+      timePrefFile << "operation,object,startTime,endTime,duration,threadID,objectSize" << endl;
+    }
+    if (!threadPrefFile.is_open()) {
+      threadPrefFile.open(thread.str().c_str());
+      //TODO: pavel fill in header line
+    }
+}
+
+void ioPostProcess()
+{
+    /*
+        Do IO post processing and output:
+            - total number of write requests (number of lines in file)
+            - throughput: (size/duration) * 1024 = MbPS
+
+        Also look into a simple CVS parser. I have used https://github.com/ben-strasser/fast-cpp-csv-parser
+        but not sure about licensing, as this is BSD license. I can write my own, but I'd prefer to
+        not re-invent the wheel.
+    */
+
+}
+
+void closeFiles() {
```

Comment:
I think it may be simpler to manage file stream closing as part of the logger destruction.  In fact, that would happen automatically, if the file streams are members of the logger class.